smalltalksrc/VMMakerTests/VMPrimitiveTest.class.st

```diff
@@ -656,6 +1017,296 @@ VMPrimitiveTest >> testPrimitiveInstVarAtPutPutsTheValueForNonIndexable [
+VMPrimitiveTest >> testPrimitiveLessOrEqualsFailsWithTypeErrorOnFirstOperand [
+	
+	interpreter push: (memory booleanObjectOf: true).
+	interpreter push: (memory integerObjectOf: 2).
+	
+	interpreter primitiveLessOrEqual. 
+	
+	self assert: interpreter failed. 
+]
+
+{ #category : #'tests - primitiveLessOrEqual' }
+VMPrimitiveTest >> testPrimitiveLessOrEqualsFailsWithTypeErrorOnSecondOperand [
+	
+	interpreter push: (memory booleanObjectOf: true).
+	interpreter push: (memory integerObjectOf: 2).
+	
+	interpreter primitiveLessOrEqual.
+	
+	self assert: interpreter failed. 
+]
+
+{ #category : #'tests - primitiveLessThan' }
+VMPrimitiveTest >> testPrimitiveLessThanFailsWithTypeErrorOnFirstOperand [
+	
+	interpreter push: (memory booleanObjectOf: true).
+	interpreter push: (memory integerObjectOf: 2).
+	
+	interpreter primitiveLessThan. 
+	
+	self assert: interpreter failed. 
+]
+
+{ #category : #'tests - primitiveLessThan' }
+VMPrimitiveTest >> testPrimitiveLessThanFailsWithTypeErrorOnSecondOperand [
+	
+	interpreter push: (memory integerObjectOf: 2).
+	interpreter push: (memory booleanObjectOf: true).
+	
+	interpreter primitiveLessThan. 
+	
+	self assert: interpreter failed. 
+]
+
+{ #category : #'tests - primitiveLessThan' }
+VMPrimitiveTest >> testPrimitiveLessThanWithNoError1 [
+	
+	interpreter push: (memory integerObjectOf: 42).
+	interpreter push: (memory integerObjectOf: 16).
+	
+	interpreter primitiveLessThan. 
+
+	self assert: interpreter stackTop equals: (memory booleanObjectOf: false). 
+]
+
+{ #category : #'tests - primitiveLessThan' }
+VMPrimitiveTest >> testPrimitiveLessThanWithNoError2 [
+	
+	interpreter push: (memory integerObjectOf: 16).
+	interpreter push: (memory integerObjectOf: 42).
+	
+	interpreter primitiveLessThan. 
+
+	self assert: interpreter stackTop equals: (memory booleanObjectOf: true). 
+]
+
+{ #category : #'tests - primitiveLessThan' }
+VMPrimitiveTest >> testPrimitiveLessThanWithNoErrorEquals [
+	
+	interpreter push: (memory integerObjectOf: 0).
+	interpreter push: (memory integerObjectOf: 0).
+	
+	interpreter primitiveLessThan. 
+
+	self assert: interpreter stackTop equals: (memory booleanObjectOf: false). 
+]
+
+{ #category : #'tests - primitiveLessThan' }
+VMPrimitiveTest >> testPrimitiveLessThanWithNoErrorPopsOperands [
+	
+	interpreter push: (memory integerObjectOf: 42). "Marker"
+	interpreter push: (memory integerObjectOf: 1).
+	interpreter push: (memory integerObjectOf: 2).
+	
+	interpreter primitiveLessThan. 
+	
+	interpreter popStack. 
+	"The previous lines pop the result of primiviteAdd (1 2) for us to check that 1 and 2 were poped from the Stack by checking that the next value is our marker, for instance 42"
+
+	self assert: interpreter stackTop equals: (memory integerObjectOf: 42). 
+]
+
+{ #category : #'tests - primitiveMod' }
+VMPrimitiveTest >> testPrimitiveMod [
+
+	interpreter push: (memory integerObjectOf: 16).
+	interpreter push: (memory integerObjectOf: 2).
```

Comment:
Would be nice to have test with something else than power of 2?

We should also have tests with negative integers, ) and floats


10 \\ -3 
-10\\ 3